internal/services/mssql/mssql_server_microsoft_support_auditing_policy_resource.go

```diff
@@ -0,0 +1,223 @@
+package mssql
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v5.0/sql"
+	"github.com/gofrs/uuid"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceMsSqlServerMicrosoftSupportAuditingPolicy() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceMsSqlServerMicrosoftSupportAuditingPolicyCreateUpdate,
+		Read:   resourceMsSqlServerMicrosoftSupportAuditingPolicyRead,
+		Update: resourceMsSqlServerMicrosoftSupportAuditingPolicyCreateUpdate,
+		Delete: resourceMsSqlServerMicrosoftSupportAuditingPolicyDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ServerMicrosoftSupportAuditingPolicyID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"server_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ServerID,
+			},
+
+			"enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"storage_endpoint": {
```

Comment:
should this be
```suggestion
			"storage_account_endpoint": {
```

Reply:
`blob_storage_endpoint` would be an even more specific name, good catch anyways! 👍🏽 

The only argument against it would be that `azurerm_mssql_server_microsoft_support_auditing_policy` is basically a copy of `azurerm_mssql_server_extended_auditing_policy`. For now I'll change it and add a TODO to that resource to bring them in sync later (`4.0`?).